File: android/PlayingCard/app/src/main/java/com/example/playingcard/data/PlayerData.kt
package com.example.playingcard.data

import kotlin.properties.Delegates

class PlayerData {
    private var cards = ArrayList<Card>()

    val numberOfCards:Int  get(){
        return cards.size
    }

    fun addCard(card: Card){
        cards.add(card)
    }

    fun getAndRemoveFirst() : Card? {

        if (cards.isEmpty()) {
            return null
        }

        val card = cards[0]
        cards.remove(card)
        return card
    }
}

File: ios/PlayingCard/PlayingCard/data/PlayerData.swift
//
//  PlayerData.swift
//  PlayingCard
//
//  Created by iblis on 28/12/2020.
//

import Foundation

class PlayerData {
    
    private var cards:[Card] = []
    
    var numberOfCards:Int {
        return cards.count
    }
    
    func addCard(card: Card){
        cards.append(card)
    }
    
    func getAndRemoveFirst() -> Card? {
        
        if cards.count == 0 {
            return nil
        }
        
        let card = cards[0]
        cards.remove(at: 0)
        return card
    }
}
